terraso_backend/apps/graphql/schema/schema.graphql

```diff
@@ -2334,9 +2334,110 @@ input DepthDependentSoilDataUpdateMutationInput {
   soilOrganicMatterTesting: SoilIdDepthDependentSoilDataSoilOrganicMatterTestingChoices
   sodiumAbsorptionRatio: Decimal
   carbonates: SoilIdDepthDependentSoilDataCarbonatesChoices
+  siteId: ID!
+  clientMutationId: String
+}
+
+type SoilDataBulkUpdatePayload {
+  errors: GenericScalar
+  results: [SoilDataBulkUpdateResultEntry!]!
+  clientMutationId: String
+}
+
+type SoilDataBulkUpdateResultEntry {
+  siteId: ID!
+  result: SoilDataBulkUpdateResult!
+}
+
+union SoilDataBulkUpdateResult = SoilDataBulkUpdateSuccess | SoilDataBulkUpdateFailure
+
+type SoilDataBulkUpdateSuccess {
+  site: SiteNode!
+}
+
+type SoilDataBulkUpdateFailure {
+  reason: SoilDataBulkUpdateFailureReason!
+}
+
+enum SoilDataBulkUpdateFailureReason {
+  DOES_NOT_EXIST
+  NOT_ALLOWED
+}
```

Comment:
This seems like a rather niche case that maybe we don't care about, but just a thought: 
If this is on a per-site level: is there ever a case where some edits should be allowed, but others not? Like, you're a manager, you edit X and Y, someone changes you to a contributor (who is allowed to edit X but not Y), and then you push?

Reply:
currently there's no difference in how a manager and contributor can modify soil data (as opposed to sites and projects), so i think this niche case does not exist as things stand